File: Chapter04/Python/config_render.py
from jinja2 import Environment, FileSystemLoader
import yaml

env = Environment(loader=FileSystemLoader("."))
template = env.get_template("cisco_template_python.txt")

with open("router_definitions.yaml") as f:
    routers = yaml.safe_load(f)

for router in routers:
    router_conf = router["name"] + "_router_config.txt"
    with open(router_conf, "w") as f:
        f.write(template.render(router))

File: Chapter04/Go/config_render.go
package main

import (
	"io/ioutil"
	"os"
	"text/template"

	"gopkg.in/yaml.v3"
)

type Router struct {
	Id     int    `yaml:"id"`
	Name   string `yaml:"name"`
	Toname string `yaml:"to_name"`
}

type RouterList []Router

func check(e error) {
	if e != nil {
		panic(e)
	}
}

func main() {
	var routers RouterList

	yamlFile, err := ioutil.ReadFile("router_definitions.yaml")
	check(err)

	err = yaml.Unmarshal(yamlFile, &routers)
	check(err)

	templateFile, err := ioutil.ReadFile("cisco_template_go.txt")
	check(err)

	for _, router := range routers {

		outFile, err := os.Create(router.Name + "_router_config.txt")
		check(err)

		tmpl, err := template.New("render").Parse(string(templateFile))
		check(err)

		err = tmpl.Execute(outFile, router)
		check(err)
	}
}
